### PR TITLE
Support wlr output management

### PR DESCRIPTION
**Pull request by wineee**

### CODE REVIEW

File: src/server/kernel/woutput.h [hunk 1]
```diff
@@ -45,6 +46,9 @@ class WAYLIB_SERVER_EXPORT WOutput : public QObject, public WObject
     Q_PROPERTY(Transform orientation READ orientation NOTIFY orientationChanged)
     Q_PROPERTY(float scale READ scale NOTIFY scaleChanged)
     Q_PROPERTY(bool forceSoftwareCursor READ forceSoftwareCursor WRITE setForceSoftwareCursor NOTIFY forceSoftwareCursorChanged)
+    Q_PROPERTY(int x READ x WRITE setX NOTIFY xChanged)
```
Comment (zccrs) [inline src/server/kernel/woutput.h:49]: Why? The output's position is control in QWOutputLayout, Don't set it in WOutput, you can control position by `OutputItem`.

**Review by zccrs — CHANGES_REQUESTED** [review]

File: src/server/kernel/woutput.h [hunk 2]
```diff
@@ -45,6 +49,7 @@ class WAYLIB_SERVER_EXPORT WOutput : public QObject, public WObject
     Q_PROPERTY(Transform orientation READ orientation NOTIFY orientationChanged)
     Q_PROPERTY(float scale READ scale NOTIFY scaleChanged)
     Q_PROPERTY(bool forceSoftwareCursor READ forceSoftwareCursor WRITE setForceSoftwareCursor NOTIFY forceSoftwareCursorChanged)
+
```
Comment (zccrs) [inline src/server/kernel/woutput.h:52]: ？

**Review by zccrs — CHANGES_REQUESTED** [review]

File: src/server/kernel/woutput.h [hunk 3]
```diff
@@ -112,6 +125,9 @@ class WAYLIB_SERVER_EXPORT WOutput : public QObject, public WObject
     void scaleChanged();
     void forceSoftwareCursorChanged();
     void bufferCommitted();
+    void requestTransform(Transform transform);
```
[the remark this answers is not in the document]
Comment (wineee) [reply src/server/kernel/woutput.h]: removed

File: src/server/qtquick/private/wquickbackend_p.h [hunk 4]
```diff
@@ -34,6 +35,7 @@ class WAYLIB_SERVER_EXPORT WQuickBackend : public WQuickWaylandServerInterface,
     explicit WQuickBackend(QObject *parent = nullptr);
 
     QW_NAMESPACE::QWBackend *backend() const;
+    Q_INVOKABLE WBackend *wbackend() const;
```
[the remark this answers is not in the document]
Comment (wineee) [reply src/server/qtquick/private/wquickbackend_p.h:38]: Yes, don,t need this now

**Review by zccrs — CHANGES_REQUESTED** [review]

File: examples/tinywl/OutputDelegate.qml [hunk 5]
```diff
@@ -262,4 +262,17 @@ OutputItem {
...
+    function setTransform(transform) {
+        onscreenViewport.rotationOutput(transform)
+    }
+
+    function setScale(scale) {
+        onscreenViewport.setOutputScale(scale)
+    }
+
+    function setOutputPosition(x, y) {
+        waylandOutput.OutputItem.item.x = x;
```
Comment (zccrs) [inline examples/tinywl/OutputDelegate.qml:275]: The root item is `OutputItem`, maybe you can using it.

**Review by zccrs — APPROVED** [review]

**Pull request merged**